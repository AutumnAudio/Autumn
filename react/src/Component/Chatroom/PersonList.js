import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Person from './Person';
const styles = {
    personListDiv: {
        marginTop: '30px',
    }
};

const PersonList = (props) => {
<<<<<<< HEAD
    const { classes, participants } = props;
    return (
        <div className={classes.personListDiv}>
            {participants.map((participant, index) => {
                return (<Person key={participant.username} person={participant}/>)
            })}
            
        </div>
=======
    const { classes } = props;
    return (
        <div className={classes.personListDiv}><Person /></div>
>>>>>>> add react prototype
    )
    
}

PersonList.propTypes = {
    classes: PropTypes.object.isRequired,
<<<<<<< HEAD
    participants: PropTypes.array.isRequired,
=======
>>>>>>> add react prototype
};

export default withStyles(styles)(PersonList)