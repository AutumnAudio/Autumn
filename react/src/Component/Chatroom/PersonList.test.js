import React from 'react'
import renderer from 'react-test-renderer'
import PersonList from './PersonList'

test('should be able to render people list', () => {
    const participantsMock = [{username: 'testUser',
                        recentlyPlayed: [{name: 'song name', artists: ['testArtist']}]},
                        {username: 'testUser2',
                        recentlyPlayed: [{name: 'another song', artists: ['testArtist2']}]}]
    const component = renderer.create(
        <PersonList participants={participantsMock} />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
