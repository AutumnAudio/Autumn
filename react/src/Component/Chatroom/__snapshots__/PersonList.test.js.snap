// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render people list 1`] = `
<div
  className="PersonList-personListDiv-1"
>
  <div
    className="Person-personDiv-5"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root Person-personIcon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
    <div
      className="Person-nameDiv-3"
    >
      testUser
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root Person-addIcon-6"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <div
      className="Person-currentSongDiv-4"
    >
      song name
       - 
      testArtist
    </div>
  </div>
  <div
    className="Person-personDiv-5"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root Person-personIcon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
    <div
      className="Person-nameDiv-3"
    >
      testUser2
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root Person-addIcon-6"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <div
      className="Person-currentSongDiv-4"
    >
      another song
       - 
      testArtist2
    </div>
  </div>
</div>
`;
